File: jscomp/test/polyvar_convert.ml
type t = 
  [ `a [@as "x"]
  | `b ]


let revData = [%raw {| {"x":"a","b":"b"} |}]  
let data = [%raw {| {"a":"x","b":"b"} |}]  
external get : 'a -> 'b -> 'c = "" [@@bs.get_index]

let tToJs (x : t ) : string = 
  get data x 

let vFromJsOpt (s : string) : t option = 
  get revData s 

let vFromJs  (s : string) : t = 
  get revData s   

File: jscomp/test/polyvar_convert.js
'use strict';


var revData = {"x":"a","b":"b"};

var data = {"a":"x","b":"b"};

function tToJs(x) {
  return data[x];
}

function vFromJsOpt(s) {
  return revData[s];
}

function vFromJs(s) {
  return revData[s];
}

exports.revData = revData;
exports.data = data;
exports.tToJs = tToJs;
exports.vFromJsOpt = vFromJsOpt;
exports.vFromJs = vFromJs;
/* No side effect */
